wl-js-analytics-adapter.sublime-snippet

```diff
@@ -0,0 +1,113 @@
+<snippet>
+  <content><![CDATA[
+/**
+ * @namespace WEBLINC.${1:adapterName}
+ */
+WEBLINC.analytics.registerAdapter('${1:adapterName}', function () {
+    'use strict';
+
+    var pageViewSent = false,
+
+        send = function (event, payload) {
+            if (WEBLINC.analytics.debug) {
+                console.log(arguments);
+            }
+
+            ${2://add your send logic here}
+        },
+
+        isLoaded = function () {
+            var scripts = _.toArray(document.scripts),
+
+            loaded = _.any(scripts, function (script) {
+                return script.src.match(WEBLINC.config.${1:adapterName}.scriptRegex);
```

Comment:
I guess it depends what the adapter is doing, and what information is required for the integration. My thought here was that there may be API keys, APP ID etc. I used a config for the personalized messaging / free shipping thing and thought it was a good idea. Happy to remove it if that's the consensus.

I'd like to include support for currentUser, I'm will check out how others have implemented it to find the nicest solution. Is there a 'best practice' way to access currentUser?